### PR TITLE
Add simple examples of gem usage

### PR DESCRIPTION
**Pull request by zzaakiirr**
### What does this PR do?

This PR adds simple examples of gem usage

### Screenshots

Screenshot 2024-07-12 at 18 18 01

## Summary by CodeRabbit

- **New Features**
  - Introduced new methods for file downloading and movement within `DownloadAndMoveFile`, `DownloadFile`, and `MoveFile` modules.
  - Added a `FailingTask` class to handle error scenarios.

- **Refactor**
  - Modified `UberTask.run` method calls to pass the `name` parameter as a positional argument instead of a named argument for improved consistency.

### CODE REVIEW

File: examples/download_and_move_file.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          )
+        end
+
+        UberTask.on_subtask_error do
+          UberTask.logger.info(
+            'Unexpected subtask error can be caught on top-level task ' \
+            'and an early return can be made, ' \
+            'isn\'t that cool!?'.underline,
+          )
+          return # Try to remove this line and see what happens!
```
Comment (borela) [inline examples/download_and_move_file.rb:25]: Remove the return, returns inside this block will affect the method that handles the [exection](https://github.com/shakacode/uber_task/blob/main/lib/uber_task/task_context.rb#L73).

We have to raise [RetryTask](https://github.com/shakacode/uber_task/blob/main/lib/uber_task.rb#L76), [SkipTask ](https://github.com/shakacode/uber_task/blob/main/lib/uber_task.rb#L126)or [EventHandled](https://github.com/shakacode/uber_task/blob/main/lib/uber_task/task_context.rb#L121C14-L121C26). 

It looks like I forgot to add a UberTask.event_handled method, add that method.

File: examples/download_file.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          )
+        end
+
+        save_external_file!(source: source, to: to)
+      end
+    end
+
+    def self.save_external_file!(source:, to:)
+      URI.parse(source).open { |io| File.write(to, io.read) }
+    rescue OpenURI::HTTPError => err
```
Comment (borela) [inline examples/download_file.rb:39]: Move this check to the parent task on_subtask_error; all errors bubble up to the parent, like in Javascript and if the parent asks the task to be retried, the current task (child) will run again.

This was the main idea to deploy apps on popmenu, we have a single place for common network errors and tasks can have many subtasks but all of them have common network errors and each time they happen, the on_subtask_error of the top task checks the error type and just ask to retry.

File: examples/download_and_move_file.rb (hunk 1 above)
Comment (zzaakiirr) [reply examples/download_and_move_file.rb]: Removed the return and added handling different types of exceptions which can be raised by subtasks

File: examples/download_and_move_file.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        # Notice that we are passing `vital: true` option so that
+        # an early return can be made if task fails
+        begin
+          Examples::DownloadFile.run(
+            source: source,
+            to: download_path,
+            vital: true,
+          )
+        rescue StandardError => err
```
[the remark this answers is not in the document]
Comment (zzaakiirr) [reply examples/download_and_move_file.rb]: Moved error check to `on_subtask_error`

File: examples/download_file.rb (hunk 2 above)
Comment (zzaakiirr) [reply examples/download_file.rb]: Moved `rescue` from here to parent task's `on_subtask_error`

**Review by zzaakiirr — COMMENTED** [review]
@borela Thanks for review! Applied your suggestions and simplified examples, can you take another look?

**Review by borela — APPROVED** [review]

**Pull request merged**